tests/python/unittest/test_operator.py

```diff
@@ -235,6 +235,55 @@ def test_regression():
                      lambda x, y : x - y)
 
 
+def check_softmax_grad():
+    x = mx.sym.Variable('x')
+    label = mx.sym.Variable('label')
+    x_nd = mx.nd.array([[1, 6, 4, 2]], ctx=default_context())
+    grad_x = mx.nd.zeros((1,4), ctx=default_context())
+    label_nd = mx.nd.array([1], ctx=default_context())
+
+    sym = mx.sym.SoftmaxOutput(data=x, label=label, ignore_label=0, use_ignore=False)
+    ex = sym.bind(ctx=default_context(), args={'x': x_nd, 'label': label_nd}, args_grad={'x': grad_x})
+
+    ex.forward(is_train=True)
+    softmax_out = ex.outputs[0].asnumpy()
+    expected_softmax_out = mx.nd.softmax(x_nd).asnumpy()
```

Comment:
it'd probably make sense to computed the expected softmax by hand. Who is to say that mx.nd.softmax and mx.sym.SoftMax don't share any code under the hood. (I guess they don't but they might in the future)

Reply:
Good idea, will do.